src/compiler/checker.ts

```diff
@@ -25284,6 +25284,55 @@ namespace ts {
                         }
                     }
                 }
+                if (type && type!==anyType && isPropertyAccessExpression(callExpression.expression)){
+                    // if (myDebug){
+                    //     consoleLog(`narrowTypeByCallExpression[dbg,in]: ==== `);
+                    // }
+                    const tmpSymbol = getSymbolOfNameOrPropertyAccessExpression(callExpression.expression);
+                    const declaration = tmpSymbol ? tmpSymbol.declarations? tmpSymbol.declarations[0] : undefined : undefined;
+                    const funcNode = declaration && (declaration as any).type
+                        && (declaration as any).type.kind === SyntaxKind.FunctionType
+                        ? (declaration as any).type as FunctionDeclaration : undefined;
+                    const funcRtnType = funcNode?.type ? getTypeOfNode(funcNode.type) : undefined;
+                    if (funcRtnType) {
+                        const antecedentResultFalse = narrowType(type, callExpression.expression, /* assumeTrue */ false);
+                        const antecedentResultTrue = narrowType(type, callExpression.expression, /* assumeTrue */ true);
+                        let hasFalsyType = false;
+                        let hasTruthyType = false;
+                        let hasDualType = false;
+
+                        // The meaning of falsy/truthy should change when the the call is the left child of a parent nullish coalescing operator `??`
+                        const nullishCoalescingOp = isBinaryExpression(callExpression.parent) && callExpression.parent.operatorToken.kind===SyntaxKind.QuestionQuestionToken;
+                        let typeOut: Type | undefined;

```

Comment:
I wrote a test case to to set the branch where `nullishCoalescingOp` is true.  However, that case seemed to be routed first through  `narrowTypeByOptionality` and never ended up calling `narrowTypeByCallExpression`.  From a software design perspective, that seems wrong in the sense that it makes it harder to verify and maintain the code.   However, the result of the test case was correct - "working as intended".  For now I'm leaving this untested branch in here as is - perhaps it would be better to code an  excepting assertion that `nullishCoalescingOp` is false?  @ahejlsberg what do you think?